src/H5Ocache.c

```diff
@@ -1268,6 +1268,11 @@ H5O__chunk_deserialize(H5O_t *oh, haddr_t addr, size_t chunk_size, const uint8_t
         UINT16DECODE(chunk_image, mesg_size);
         if (mesg_size != H5O_ALIGN_OH(oh, mesg_size))
             HGOTO_ERROR(H5E_OHDR, H5E_CANTLOAD, FAIL, "message not aligned");
+        if (mesg_size > (size_t)(p_end - chunk_image))
```

Comment:
I believe the later check for "Try to detect invalidly formatted object header message that extends past end of chunk." is now obsolete (and it had an off by one error that may have caused the problem in the first place) and can be deleted. I also think having both p_end and eom_ptr doing almost the same thing is confusing and redundant but it doesn't need to be fixed now.